app/src/main/kotlin/cz/covid19cz/erouska/exposurenotifications/worker/ProcessDiagnosisKeysWorker.kt

```diff
@@ -0,0 +1,62 @@
+package cz.covid19cz.erouska.exposurenotifications.worker
+
+import android.content.Context
+import androidx.work.*
+import cz.covid19cz.erouska.exposurenotifications.ExposureNotificationsRepository
+import org.koin.core.KoinComponent
+import org.koin.core.inject
+
+/**
+ * Worker that downloads and processes diagnosis keys.
+ */
+class ProcessDiagnosisKeysWorker(
+    appContext: Context,
+    workerParams: WorkerParameters
+) : CoroutineWorker(appContext, workerParams), KoinComponent {
+
+    companion object {
+        private const val TAG = "ProcessDiagnosisKeysWorker"
+
+        private const val ARGUMENT_TOKEN = "token"
+
+        /**
+         * Enqueue download and processing diagnosis keys.
+         */
+        fun enqueueProcessingOfDiagnosisKeys(workManager: WorkManager, token: String) {
+            val constraints = Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED) // internet access
+                .build()
+
+            val request = OneTimeWorkRequestBuilder<ProcessDiagnosisKeysWorker>()
+                .setConstraints(constraints)
+                .setInputData(
+                    Data.Builder()
+                        .putString(ARGUMENT_TOKEN, token)
+                        .build()
+                )
+                .addTag(TAG)
+                .build()
+
+            workManager.enqueueUniqueWork(
+                TAG,
+                ExistingWorkPolicy.REPLACE,
+                request
+            )
+        }
+    }
+
+    private val exposureNotificationsRepository: ExposureNotificationsRepository by inject()
+
+    override suspend fun doWork(): Result {
+        val token = inputData.getString(ARGUMENT_TOKEN)
+        try {
+            token?.let {
+                //exposureNotificationsRepository.provideDiagnosisKeys(token)
```

Comment:
Tohle má stahovat ZIPy? Sem patří nějaké TODO asi?

Reply:
to je nepoužitá třída z experimentování

Reply:
Takže smazat tu zakomentovanou řádku?

Reply:
je to work in progress, celá ta třída se bude ještě dost měnit, to neřeš :)